charts/nx-cloud/templates/ingress.yaml

```diff
@@ -0,0 +1,68 @@
+{{- if .Values.ingress.enabled -}}
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: nx-cloud-ingress
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "nxCloud.mergedLabels" (dict "resourceLabels" .Values.ingress.labels "context" .) | nindent 4 }}
+  {{- with .Values.ingress.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  {{- if .Values.ingress.className }}
+  ingressClassName: {{ .Values.ingress.className }}
+  {{- end }}
+  {{- if .Values.ingress.tls.enabled }}
+  tls:
+    - secretName: {{ .Values.ingress.tls.secretName }}
+      {{- if .Values.ingress.tls.hosts }}
+      hosts:
+        {{- range .Values.ingress.tls.hosts }}
+        - {{ . | quote }}
+        {{- end }}
+      {{- else }}
+      hosts:
+        - {{ .Values.ingress.host | quote }}
+      {{- end }}
+  {{- end }}
+  rules:
+    - host: {{ .Values.ingress.host }}
+      http:
+        paths:
+          - path: /
+            pathType: Prefix
+            backend:
+              service:
+                name: {{ .Values.frontend.service.name }}
+                port:
+                  number: {{ .Values.frontend.service.port }}
+          - path: /nx-cloud
+            pathType: Prefix
+            backend:
+              service:
+                name: {{ .Values.api.service.name }}
+                port:
+                  number: {{ .Values.api.service.port }}
+          {{- if .Values.fileServer.enabled }}
+          - path: /file
+            pathType: Prefix
+            backend:
+              service:
+                name: {{ .Values.fileServer.service.name }}
+                port:
+                  number: {{ .Values.fileServer.service.port }}
+          {{- end }}
+          {{- if .Values.ingress.additionalPaths }}
+          {{- range .Values.ingress.additionalPaths }}
+          - path: {{ .path }}
+            pathType: {{ .pathType }}
+            backend:
+              service:
+                name: {{ .service.name }}
+                port:
+                  number: {{ .service.port }}
+          {{- end }}
```

Comment:
for this, why decompose the object? we could just take a literal array of these path specifications and then just yaml/indent them no?